src/lib/include/libsed.h

```diff
@@ -185,6 +202,17 @@ int sed_revertlsp(struct sed_device *dev, const struct sed_key *key, bool keep_g
 int sed_setpw(struct sed_device *dev, const struct sed_key *old_key,
 		const struct sed_key *new_key);
 
+/**
+ * Get list of locking ranges.
+ * @param dev			the device to operate on
+ * @param key   		the Admin1 password
+ * @param lrs			array of discovered locking ranges
+ *
+ * @return OPAL_SUCCESS on success
+ */
+int sed_list_lr(struct sed_device *dev, const struct sed_key *key,
+                struct sed_opal_lockingrange_arr *lrs);
```

Comment:
it is good programming practice when passing pointer to first element of array to also pass its size, so function can check if data has not been accessed outside of array boundary

Reply:
lrs is a pointer to structure not to first element. Maybe name of this structure is confusing, so I changed it to sed_opal_lockingranges.

Reply:
correct -- this is a structure not an array. 
but problem exists in list_lr() function in nvme_pt_ioctl.c
lrs->lr_num can be set to number greater than SED_OPAL_MAX_LRS, e.g. 10
later in that same function you loop:
for (int i = 0; i < lrs->lr_num; i++) {
....
//take reference to item in the array
struct sed_opal_lockingrange *lr = &lrs->lrs[i];
// and update that item
lr->lr_id = i;
}

that loop will for i>=9 will start writing outside of array boundary. this needs to be fixed

Reply:
I think the issue has been resolved in the newest version by limiting lrs->lr_num to SED_OPAL_MAX_LRS.